crates/hc/src/lib.rs

```diff
@@ -108,19 +108,54 @@
 //! ```
 //! and the examples.
 
+use std::process::Command;
+
 // Useful to have this public when using this as a library.
 pub use holochain_cli_bundle as hc_bundle;
 use holochain_cli_sandbox as hc_sandbox;
-use structopt::StructOpt;
+use structopt::{lazy_static::lazy_static, StructOpt};
+
+mod external_subcommands;
+
+lazy_static! {
+    static ref HELP: &'static str = {
+        let extensions = external_subcommands::list_external_subcommands()
+            .into_iter()
+            .map(|s| format!("    hc {}\t  Run \"hc {} help\" to see its help", s, s))
+            .collect::<Vec<String>>()
+            .join("\n");
+
+        let extensions_str = match extensions.len() {
+            0 => format!(""),
+            _ => format!(
+                r#"
+CLI EXTENSIONS:
+{extensions}"#
+            ),
+        };
+
+        let s = format!(
+            r#"Holochain CLI
+
+Work with DNA, hApp and web-hApp bundle files, set up sandbox environments for testing and development purposes, make direct admin calls to running conductors, and more.
+{extensions_str}"#
+        );
+        Box::leak(s.into_boxed_str())
+    };
+}
+
+fn builtin_commands() -> Vec<String> {
+    ["hc-web-app", "hc-dna", "hc-app", "hc-sandbox"]
```

Comment:
Would be nice if there were a way to keep these in sync with our actual built-in commands so we don't have to remember to update this when we add a new one. But not a huge deal.

Also this makes me wish we had chosen the other order for pack/unpack, so `hc pack dna` instead of `hc dna pack`.

Reply:
Yeah I tend to agree... I don't know of a good way to get the binary names from a dependency crate... buildinfo maybe?

Reply:
I wonder if we should (now or eventually) just move the built-in commands to their own binaries too? Then we don't have to distinguish between them. What do you think?